### ISSUE TITLE
crash symbol_names in backtrace contents may lost sometimes when  packaged by xcode 14 series

### ISSUE THREAD

**Issue by lyandy** [getsentry/sentry-cocoa#2530]
### Platform

iOS

### Installed

CocoaPods

### Version

7.30.1

### Steps to Reproduce

I have created one demo to reproduce it stably.
ref: https://github.com/lyandy/sentry_text

1. download demo and pod install
2. launch project and run
3. click __make crash__
4. find and open sentry crash reports folder
5. open crash json file, located to the crashed content in backtraces
6. search the  object name __sentry_test__ 

### Expected Result

we can get the corrected stack. like `-[xxx, xxx]`

### Actual Result

If you use Xcode 14.1 or Xcode 14.2 to run this demo and make crash，you will find that `symbol_name` is `""`.😲
But if you use Xcode 13.4.1, you can see that `symbol_name` has recorded the correct stack `-[ViewController crash:]` 😁

The first time i saw this condition was last week about 2022.12.6. 
And that day, i update the xcode on my packaged device from 13.4.1 to 14.1 😂

**Comment by philipphofmann** [comment]
Thanks for reporting this @lyandy. I just tried this with a simulator and Xcode 14, and I still get `symbol_name`.

I'm curious about why you need to access the SentryCrash report. Can you elaborate on why you need this, please? Sentry should still be able to properly symbolicate your crashes and show you the proper function names.

**Comment by lyandy** [comment]
> Thanks for reporting this @lyandy. I just tried this with a simulator and Xcode 14, and I still get `symbol_name`.
> 
> I'm curious about why you need to access the SentryCrash report. Can you elaborate on why you need this, please? Sentry should still be able to properly symbolicate your crashes and show you the proper function names.

hi @philipphofmann 
I tried a simulator with the latest Xcode version 14.2, `symbol_name` is still `""` which `object_name` is __sentry_test__. Maybe , u update Xcode to 14.1 or 14.2, it can be reproduced.

[sentry_test-report-0000000048800000.json](https://github.com/lyandy/sentry_text/blob/main/sentry_test-report-0000000048800000.json)

 And I figured out how `""` looks like on the senty-host
<img width="873" alt="image" src="https://user-images.githubusercontent.com/3918529/208021995-57918b5c-1459-4dad-91fc-3d1272600d15.png">

Perhaps, not long time from now,  my sentry-host will be filled with the meaningless stacks. 


Since sentry can not provide the crash crate for an app, so i develop a tool to analyze the crash details with secondary development base on sentry. 

Everytime, before sentry deleted the SentryCrash report, my tool would analyze the data structure of crash report , and report some key point stacks to my clichhouse server. 

As a result, i found that crash `symbol_name` in backtrace contents may be  `""`  sometimes when packaged by xcode 14 series

**Comment by philipphofmann** [comment]
> Since sentry can not provide the crash crate for an app

What is the crash crate? Are you referring to the crash rate? If yes, we have [Release Health](https://docs.sentry.io/product/releases/health/#crash), which does precisely that.

**Comment by lyandy** [comment]
> > Since sentry can not provide the crash crate for an app
> 
> What is the crash crate? Are you referring to the crash rate? If yes, we have [Release Health](https://docs.sentry.io/product/releases/health/#crash), which does precisely that.

ok, i see it. thanks @philipphofmann . [Release Health](https://docs.sentry.io/product/releases/health/#crash) can actually reflect the crash rate.

Uh.. let's back to the issue, do u have any idea about why crash stack `symbol_name` is "" when package by Xcode 14.1 or 14.2

**Comment by philipphofmann** [comment]
> do u have any idea about why crash stack symbol_name is "" when app packaged by Xcode 14.1 or 14.2

No, I need to investigate that, but I can't give you an ETA for that.
